examples/hmm.py

```diff
@@ -22,6 +22,7 @@
 
 import torch
 import torch.nn as nn
+from torch.distributions import constraints
```

Comment:
Hmm, I guess since the expectation is inside the log, this might be better called "collapsed SVI" or "collapsed SGVB". There are no sampled latent variables, and the only stochasticity is in data subsampling. @martinjankowiak I defer to your judgement as to what to call these algorithms.